conans/client/graph/graph_binaries.py

```diff
@@ -291,17 +293,19 @@ def _compute_package_id(self, node, default_package_id_mode, default_python_requ
         # A bit risky to be done now
         conanfile = node.conanfile
         neighbors = node.neighbors()
+
+        direct_reqs, indirect_reqs = self.package_id_transitive_reqs(node)
```

Comment:
My strategy here is to use always the new method (the good one) to avoid duplicating logic and then I'm introducing the conan-v1-bug for backward compatibility.